src/Network/NWProtocolIPOptions.cs

```diff
@@ -0,0 +1,115 @@
+//
+// NWProtocolTls: Bindings the Netowrk nw_protocol_options API focus on TLS options.
+//
+// Authors:
+//   Manuel de la Pena <mandel@microsoft.com>
+//
+// Copyrigh 2018 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+using Security;
+using OS_nw_protocol_definition=System.IntPtr;
+using IntPtr=System.IntPtr;
+
+namespace Network {
+
+	[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+	public enum NWIPLocalAddressPreference {
+		Default = 0,
+		Temporary = 1,
+		Stable = 2,
+	}
+
+	public enum NWIPVersion {
+		Any = 0,
+		Version4 = 1,
+		Version6 = 2,
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public class NWProtocolIPOptions : NWProtocolOptions {
+		// default values, same are uset ins swift
+		NWIPVersion version = NWIPVersion.Any;
+		nuint hopLimit = (nuint) 0;
+		bool useMinimumMtu = false;
+		bool disableFragmentation = false;
+		bool shouldCalculateReceiveTime = false;
+		NWIPLocalAddressPreference localAddressPreference = NWIPLocalAddressPreference.Default;
+
+
+		internal NWProtocolIPOptions (IntPtr handle, bool owns) : base (handle, owns) {}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ip_options_set_version (IntPtr options, NWIPVersion version);
+
+		public NWIPVersion Version {
+			get => version;
```

Comment:
that's tricky and I'm not sure it works - i.e. it only works if 
* the managed setter is used for updates;
* the .ctor(IntPtr) is only used to create default instance

IOW if you have a handle to a native instance then you'll be initializing the managed instance with incorrect data.